nucypher/blockchain/eth/actors.py

```diff
@@ -1202,6 +1206,26 @@ def missing_commitments(self) -> int:
         missing = self.staking_agent.get_missing_commitments(checksum_address=staker_address)
         return missing
 
+    def verify_policy_payment(self, hrac: bytes) -> None:
+        arrangements = self.policy_agent.fetch_policy_arrangements(policy_id=hrac)
+        members = set()
```

Comment:
It took me a while to understand what's happening here - we populate a set only to later use it to determine error message to supply?

Can we just use a `frozenset` here instead?

Better yet, why not just use a sentinel for the candidate name (ie, `arrangement = None`) and then check that?

Reply:
Not actually added in this PR, it's from `chili-con-carne` on which this PR is based